lib/plugins/bed.js

```diff
@@ -10,35 +10,64 @@ function noop (err) {
 
 function inject (bot) {
   bot.isSleeping = false
+  const mcData = require('minecraft-data')(bot.version)
+
   const beds = new Set(['white_bed', 'orange_bed', 'magenta_bed', 'light_blue_bed', 'yellow_bed', 'lime_bed', 'pink_bed', 'gray_bed',
     'light_gray_bed', 'cyan_bed', 'purple_bed', 'blue_bed', 'brown_bed', 'green_bed', 'red_bed', 'black_bed', 'bed'])
 
+  const hostileMobs = new Set(['blaze', 'creeper', 'drowned', 'endermite', 'enderman', 'evoker', 'giant', 'guardian', 'elder_guardian',
+    'illusioner', 'pillager', 'phantom', 'ravager', 'silverfish', 'skeleton', 'stray', 'wither_skeleton', 'spider', 'cave_spider', 'vex',
+    'vindicator', 'witch', 'Wither', 'zombie', 'zombie_villager', 'husk'])
+
   function isABed (block) {
     return beds.has(block.name)
   }
 
   function parseBedMetadata (bedBlock) {
-    const bitMetadata = bedBlock.metadata.toString(2).padStart(4, '0')
-
     const metadata = {
-      part: parseInt(bitMetadata[0], 2),
-      occupied: parseInt(bitMetadata[1], 2),
-      facing: 0,
+      part: false, // true: head, false: foot
+      occupied: 0,
+      facing: 0, // 0: south, 1: west, 2: north, 3 east
       headOffset: new Vec3(0, 0, 1)
     }
 
-    switch (bitMetadata.slice(2, 4)) {
-      case '01':
-        metadata.facing = 1
-        metadata.headOffset.set(-1, 0, 0)
-        break
-      case '10':
-        metadata.facing = 2
-        metadata.headOffset.set(0, 0, -1)
-        break
-      case '11':
-        metadata.facing = 3
-        metadata.headOffset.set(1, 0, 0)
+    if (bot.majorVersion === '1.13') {
+      const state = bedBlock.stateId - mcData.blocksByStateId[bedBlock.stateId].minStateId
+      const bitMetadata = state.toString(2).padStart(4, '0') // FACING (2 bits) | PART (1 bit) | OCCUPIED (1 bit)
+      metadata.part = !parseInt(bitMetadata[3], 2)
+      metadata.occupied = !parseInt(bitMetadata[2], 2)
```

Comment:
As those are single characters, couldn't you just compare like `bitMetadata[3] === '0'`?